### DISCUSSION TITLE
Handle submit for form in Wizard with multiple steps

### DISCUSSION THREAD

**Discussion by thaorell** [react-hook-form/react-hook-form#7504] [Q&A] [answered]
I need to trigger the `handleSubmit(onSubmit)` of a nested form inside a Wizard component from a UI library with a global Next button.

**Wizard.jsx**
```
const Wizard = (props) => {
 const [formValues, setFormValues] = useState({});
 const onSubmit = data => console.log(data);
  return (
    <Wizard>
      <MyForm onSubmit={onSubmit} />
      <WizardFooter>
        <button type="submit">Next</button>
      </WizardFooter>
    </Wizard>
  );
};
 
```

**MyForm.jsx**
```
const MyForm = (props) => {
  const methods = useForm();
  return (
    <form onSubmit={methods.handleSubmit(props.onSubmit)}>
      {/*components*/}
    </form>
  );
};
```

My form will be reusable by different steps of the Wizard. My goal is to trigger `handleSubmit`, get the data from `MyForm`, then pass the data to `Wizard` in update `formValues` by clicking `Next`. And the problem is that my `Next` button is not in the `MyForm`. How can I trigger handleSubmit?

**Comment by bluebill1049** [comment]
some references that maybe helpful

- https://github.com/react-hook-form/react-hook-form/issues/7478#issuecomment-1009609294
- https://react-hook-form.com/advanced-usage#WizardFormFunnel

**Reply by thaorell** [reply]
Thank you. My situation is a bit different. In my case, the Next/Submit button of each step is defined outside of the form and in WizardFooter. How can I achieve the same effect?

**Reply by bluebill1049** [reply]
i believe if you can use `FormProvider`.

**Reply by thaorell** [reply]
Hi, I tried, to put `methods.handleSubmit(onSubmit)` in the `onClick` of `Next` but it didn't work.  In your example, the  `<input type="submit" />` in each step triggers the onSubmit. I don't have a button that fires a submit for each step. I need to trigger the submit event from its parent. Is there a way around it?

**Reply by AmilaDotDev** [reply]
Hi, I asked a similar question related to multistep form, on stackoverflow yesterday if someone can have a look please. https://stackoverflow.com/questions/70673394/using-react-hook-form-in-multistep-without-using-a-state-manager

The [codesandbox](https://codesandbox.io/s/react-hook-form-multisteps-xz4xq?file=/src/App.js) related to that question 
Cheers

**Comment by Moshyfawn** [comment, marked as answer]
[The button `form` attribute](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#attr-form) might be what you're after. You can add a form `id` to the your form element _(maybe pass it into your Wizard component as a prop, if it's abstracted)_ and add the `form` attribute to your submit button with the same form `id` you want it to trigger the submit event of.
See [my example codesandbox](https://codesandbox.io/s/button-form-attribute-example-ixekr). It's not a Wizard form, but you can easily apply this concept to whatever use-case you have.

**Reply by thaorell** [reply]
Thanks. It worked!
